### ISSUE TITLE
Jetty 9.3.11 - GzipHandler.handle() does not consider MinGzipSize

### ISSUE THREAD

**Issue by CharlesLenk** [eclipse/jetty.project#1305]
The _minGzipSize in GzipHandler is checked in the getDeflater() method, but not in the handle() method. I've stepped through with a debugger to confirm that the _minGzipSize was set correctly in my application's instance of GzipHandler, and that it's not considered as part of the handle() execution. This results in all requests to my application with the "accept-encoding: gzip" header having gzipped responses, regardless of whether it's efficient to gzip a response of that size.

**Comment by gregw** [comment]
Charles,

can you give us some example code that causes this issue?

Typically that situation is caused when a response is flushed before any indication of size is known.  For example if you wrote:
```java
    response.getOutputStream().write("X".getBytes());
    response.getOutputStream().flush();
```
Then jetty has to commit the response without knowing the length, so it will assume it is going to be large and gzip it.

If however, your app has a known small size response at the time of commit, then this could indeed be a jetty bug.  More info is needed.

**Comment by CharlesLenk** [comment]
Hi Greg,

While writing a demo app for this issue, I found that the problem I saw actually relates to how SpringBoot handles some response types. Apparently, a content-length isn't added when using the Jackson message converter, but is in a number of other cases. As you suggested, this could be related to when the output stream is flushed.

I mistook the issue for Jetty not checking the against the minGzipSize in GzipHandler, which it doesn't, but it does appear to check against it elsewhere, because I'm seeing the correct compression behavior after writing a filter to append the content length before committing the response.

Sorry for raising a non-issue.

Thanks,
-Charles
